plugins/restful/RestfulDataProviderVariable.php

```diff
@@ -0,0 +1,306 @@
+<?php
+
+/**
+ * @file
+ * Contains \RestfulDataProviderDbQuery
+ */
+
+abstract class RestfulDataProviderVariable extends \RestfulBase implements \RestfulDataProviderInterface {
+
+  /**
+   * Constructs a RestfulDataProviderVariable object.
+   *
+   * @param array $plugin
+   *   Plugin definition.
+   * @param RestfulAuthenticationManager $auth_manager
+   *   (optional) Injected authentication manager.
+   * @param DrupalCacheInterface $cache_controller
+   *   (optional) Injected cache backend.
+   * @param string $language
+   *   (optional) The language to return items in.
+   */
+  public function __construct(array $plugin, \RestfulAuthenticationManager $auth_manager = NULL, \DrupalCacheInterface $cache_controller = NULL, $language = NULL) {
+    parent::__construct($plugin, $auth_manager, $cache_controller, $language);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function controllersInfo() {
```

Comment:
Is there any need to re-define the controller names? Can't we just use the usual ones? I think it helps with setting expectation on what's in a data provider class.